typecheck/typecheck.go

```diff
@@ -948,6 +950,19 @@ func (c *Checker) exprBuiltinCall(e *expr.Call) partial {
 		return p
 	case tipe.Close:
 		p.typ = nil
+		if len(e.Args) != 1 {
+			p.mode = modeInvalid
+			c.errorf("too few arguments to close")
+			return p
+		}
+		arg := c.expr(e.Args[0])
+		_, isChan := tipe.Underlying(arg.typ).(*tipe.Chan)
+		if !isChan {
+			p.mode = modeInvalid
+			c.errorf("argument to close must be a chan, got %s", format.Type(arg.typ))
+			return p
+		}
```

Comment:
One more condition to check here, you cannot close a receive-only channel.

Reply:
done.